core/actionProxy/actionproxy.py

```diff
@@ -136,13 +136,17 @@ def error(msg):
         # run the process and wait until it completes.
         (o, e) = p.communicate()
 
+        # stdout/stderr may be either text or bytes, depending on Python
+        # version.   In the latter case, decode to text.
+        o, e = map(lambda x: x.decode() if isinstance(x, bytes) else x, (o, e))
```

Comment:
The map(lambda) makes my head spin.  Please consider changing this to the longer, but more understandable:
```python
if isinstance(o, bytes):
        o = o.decode('utf-8')
if isinstance(e, bytes):
        e = e.decode('utf-8')
```

Also line 143 (if o is not None) will _always_ be True.  o might be '' but it will _never_ be None.  Same for e on line 151.

You might consider changing these to `if o:` and `if e:` which more closely aligns with the original intention.

Finally, I not recommend all of the [u'' changes](http://python-future.org/unicode_literals.html#unicode-literals) unless you see an absolute necessity.